tests/slow/src/test/scala/tests/feature/Scala3CodeActionLspSuite.scala

```diff
@@ -328,6 +338,264 @@ class Scala3CodeActionLspSuite
     expectNoDiagnostics = false,
   )
 
+  check(
```

Comment:
We can really just add one test (first one looks good), since we only want to test the branch when we have Scala 3 dialect.  All the other cases are already handled in the ConvertToNamedArgumentsLspSuite.

Anything specific to Scala 3 we can test in cross suite.

The reason is for this comment is that these tests each can last up to a couple of seconds and it's harsh on out CI.

Reply:
Ah, right.

> All the other cases are already handled in the ConvertToNamedArgumentsLspSuite.
`ConvertToNamedArgumentsLspSuite` test against Scala 2.13, it doesn't verify anything against Scala 3.

Actually, those tests that verify the behavior (rather than the first one) should be in PC tests instead of `xxxLSPSuite`.
Sorry, we should did this in https://github.com/scalameta/metals/pull/3971

For example, Implementing all abstract member code action 
 have two (three?) test suite 
- cross test in pc (presentation compiler) 
  - https://github.com/scalameta/metals/blob/53249aa4047ee304275a818d3995a4b56a0b42b2/tests/cross/src/test/scala/tests/pc/AutoImplementAbstractMembersSuite.scala#L14
  - These tests verify the behavior of code action, but it doesn't check the code action is available or not. Those test anyway invoke the code action even if the user won't see it's available.
  - These tests run against all Scala versions
  - You can run those tests with the specific version by `++3.1.3 cross/testOnly tests.pc.AutoImplementAbstractMemberSuite`.
- ImplementAbstractMemberLspSuite
  - https://github.com/scalameta/metals/blob/53249aa4047ee304275a818d3995a4b56a0b42b2/tests/unit/src/test/scala/tests/codeactions/ImplementAbstractMembersLspSuite.scala#L5-L6
  - These test verify the code actions are available for users when we have a certain error (missing implementation error)
  - This test runs against only Scala 2.13
- Scala3CodeActionLspSuite
  - https://github.com/scalameta/metals/blob/53249aa4047ee304275a818d3995a4b56a0b42b2/tests/slow/src/test/scala/tests/feature/Scala3CodeActionLspSuite.scala#L332-L358
  - This test check the code action is available in Scala3. We test this because if the Scala3 compiler has different error message than Scala2, we might need to change the regex to capture the error message for missing implementation error.
  - This test runs against only Scala3.

---

So, we should migrate most of `ConvertToNamedArgumentsLspSuite` into pc suite (that we haven't yet implemented). but it can be done in another PR :)

Reply:
Scala3CodeActionLspSuite will run the entire server, while cross tests will only invoke the compiler, which should be much faster.

Reply:
It's probably fine to test if the workflow works for Scala 3, which confirms one new path for testing, while we can test anything specific to Scala 3 compiler in the cross tests.